rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2024,6 +2019,66 @@ rocalRawCIFAR10Source(
     return output;
 }
 
+RocalTensor ROCAL_API_CALL
+rocalJpegExternalFileSource(
+    RocalContext p_context,
+    const char* source_path,
+    RocalImageColor rocal_color_format,
+    bool is_output,
+    bool shuffle,
+    bool loop,
+    RocalImageSizeEvaluationPolicy decode_size_policy,
+    unsigned max_width,
+    unsigned max_height,
+    RocalDecoderType dec_type,
+    RocalExternalSourceMode external_source_mode) {
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try {
+        bool decoder_keep_original = (decode_size_policy == ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED) || (decode_size_policy == ROCAL_USE_MAX_SIZE_RESTRICTED);
+        DecoderType decType = DecoderType::TURBO_JPEG;  // default
+        if (dec_type == ROCAL_DECODER_OPENCV) decType = DecoderType::OPENCV_DEC;
+        if ((decode_size_policy == ROCAL_USE_MAX_SIZE) || (decode_size_policy == ROCAL_USE_MAX_SIZE_RESTRICTED))
+            THROW("use_max_size is not supported in external source reader");
+
+        // user need to specify this
+        if (max_width == 0 || max_height == 0) {
+            THROW("Invalid input max width and height");
+        } else {
+            LOG("User input size " + TOSTR(max_width) + " x " + TOSTR(max_height))
+        }
+
+        auto [width, height] = std::make_tuple(max_width, max_height);
+        auto [color_format, tensor_layout, dims, num_of_planes] = convert_color_format(rocal_color_format, context->user_batch_size(), height, width);
+        INFO("Internal buffer size width = " + TOSTR(width) + " height = " + TOSTR(height) + " depth = " + TOSTR(num_of_planes))
+
+        auto info = TensorInfo(std::move(dims),
+                               context->master_graph->mem_type(),
+                               RocalTensorDataType::UINT8,
+                               tensor_layout,
+                               color_format);
+        output = context->master_graph->create_loader_output_tensor(info);
+        context->master_graph->set_external_source_reader_flag();
+        
+        unsigned shard_count = 1;  // Hardcoding the shard count to 1 for now.
```

Comment:
Probably, we need to support multi-shards for multi-GPU training with external source. Suggest to add that as a separate PR